src/lib.rs

```diff
@@ -211,7 +225,14 @@ impl Crust {
             if our_type != OurType::Client {
                 listener::listen(incoming_connections);
             }
+
+            let _ = err_tx.send(Ok(()));
         });
+
+        err_rx
+            .recv()
+            .map_err(Error::ChannelRecv)
```

Comment:
If you already have a `from()` conversion what's the use of mapping the error - why not simply `Ok(blah()?)` kind of semantics ?

Reply:
In this case `from()` simply does not get applied to the error - don't know why though.